cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp

```diff
@@ -21,6 +21,8 @@
 
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
 
+    <c:set var="serviceUrl" value="${not empty osfLoginContext.getServiceUrl() ? osfLoginContext.getServiceUrl() : fn:escapeXml(param.service)}" />
```

Comment:
Some internal flow does not have `loginContext`, in which case we use the `service` parameter in request instead of `osfLoginContext.getServiceUrl()`. For throttle login access denied page, user can click "Sign I" to go back to CAS login page with current service.